src/network/connection.cpp

```diff
@@ -1861,16 +1807,17 @@ void ConnectionSendThread::sendPackets(float dtime)
 
 		if (dynamic_cast<UDPPeer*>(&peer)->m_pending_disconnect)
 		{
-			pendingDisconnect.push_back(*j);
+			pendingDisconnect.push_back(peerId);
 		}
 
 		PROFILE(std::stringstream peerIdentifier);
-		PROFILE(peerIdentifier << "sendPackets[" << m_connection->getDesc() << ";" << *j << ";RELIABLE]");
+		PROFILE(peerIdentifier << "sendPackets[" << m_connection->getDesc() << ";" << peerId
+			<< ";RELIABLE]");
 		PROFILE(ScopeProfiler peerprofiler(g_profiler, peerIdentifier.str(), SPT_AVG));
 
 		LOG(dout_con<<m_connection->getDesc()
-				<< " Handle per peer queues: peer_id=" << *j
-				<< " packet quota: " << peer->m_increment_packets_remaining << std::endl);
+				<< " Handle per peer queues: peer_id=" << peerId
+			<< " packet quota: " << peer->m_increment_packets_remaining << std::endl);
 		// first send queued reliable packets for all peers (if possible)
 		for (unsigned int i=0; i < CHANNEL_COUNT; i++)
 		{
```

Comment:
no we cannot use a range based for loop, but we can store the pointer instead of reuse the verbose mode on each call :p